cmd/celestia-appd/cmd/root.go

```diff
@@ -151,12 +151,16 @@ func initRootCmd(rootCmd *cobra.Command, encodingConfig encoding.Config) {
 
 	server.AddCommands(rootCmd, app.DefaultNodeHome, NewAppServer, createAppAndExport, addModuleInitFlags)
 
+	// set the default keyring backend to `file`
+	keybase := keys.Commands(app.DefaultNodeHome)
+	keybase.Flag(flags.FlagKeyringBackend).DefValue = "file"
```

Comment:
[blocking question]
do we know if this default will get overwritten by the client.toml setting? Will this default override that setting? will changing that have a different effect?

Reply:
I know this is merged, but can we follow up with these questions as well @rahulghangas ?

Reply:
My client.toml has `keyring-backend = "os"` so it seems like in increasing order of precedence:

1. client.toml
2. the default value on this line
3. CLI flag: `--keyring-backend`

In other words, `client.toml` isn't helpful for end-users because the value specified on this line will override it

Reply:
^ disregard, the help message is misleading. I did some investigating on [this branch](https://github.com/rootulp/cosmos-sdk/tree/rp/keys-debug) and learned that with this change, the help message claims the default is `file` but the actual keyring backend that gets used is whatever is in client.toml. If a value isn't specified in client.toml, then I encounter:

```error
$ celestia-appd keys debug
Error: couldn't get key ring: unknown keyring backend
```

even though that's when I would expect the default specified on this line to be applied.

Reply:
Can resolve. See https://github.com/celestiaorg/celestia-app/issues/1201